### PR TITLE
feat(yaml): more comprehensive grammar

### PR DESCRIPTION
**Pull request by dyc3**
## Summary

This PR contains a more comprehensive grammar for YAML that should make it possible to implement a proper lexer, with support for *most* language features.

related to: #2365

### CODE REVIEW

File: xtask/codegen/src/yaml_kinds_src.rs [hunk 1]
```diff
@@ -24,13 +24,31 @@ pub const YAML_KINDS_SRC: KindsSrc = KindsSrc {
...
+        "YAML_IDENTIFIER",
+    ],
     tokens: &["NEWLINE", "WHITESPACE", "IDENT", "COMMENT"],
     nodes: &[
         "YAML_ROOT",
         "YAML_DOCUMENT_LIST",
         "YAML_DOCUMENT",
-        "YAML_CONTENT_LIST",
+        "ANY_YAML_VALUE",
+        "ANY_YAML_SCALAR",
```
Comment (ematipico) [inline xtask/codegen/src/yaml_kinds_src.rs:41]: Any* aren't real nodes, so we should not add them here.

File: xtask/codegen/yaml.ungram [hunk 2]
```diff
@@ -30,14 +30,66 @@ YamlBogus = SyntaxElement*
...
 
-AnyYamlContent = YamlScalar | YamlBogusValue
-YamlScalar = value: 'yaml_scalar'
-YamlStringLiteral = value: 'yaml_string_literal'
+
+YamlNumberValue = value: 'yaml_number_value'
+YamlStringValue = value: 'yaml_string_value'
+YamlBooleanValue = value: 'yaml_boolean_value'
+YamlNullValue = value: 'yaml_null_value'
+YamlIdentifier = value: 'yaml_identifier'
```
Comment (ematipico) [inline xtask/codegen/yaml.ungram:48]: We usually keep values at the end of the file

File: xtask/codegen/yaml.ungram [hunk 3]
```diff
@@ -30,14 +30,66 @@ YamlBogus = SyntaxElement*
...
+
+YamlObjectMemberList = YamlObjectMember*
+
+YamlBlockLiteral =
+	'|'
+	value: 'yaml_block'
+
+YamlBlockFolded =
+	'>'
+	value: 'yaml_block'
```
Comment (ematipico) [inline xtask/codegen/yaml.ungram:95]: You should create a `YamlBlockLiteral`

Comment (dyc3) [reply xtask/codegen/yaml.ungram:95]: In the yaml spec, the `|` char denotes a "block literal" where newlines are included in the value, and the `>` denotes a "folded block" where newlines are ignored, so I think it should be `YamlBlockValue` instead.

Comment (ematipico) [reply xtask/codegen/yaml.ungram]: The main idea was to use the same node in both definitions. It seems they are the same

**Review by ematipico — APPROVED** [review]

**Pull request merged**